test/raft/fsm_test.cc

```diff
@@ -1565,7 +1569,7 @@ BOOST_AUTO_TEST_CASE(test_non_voter_stays_pipeline) {
     A.add_entry(newcfg);
     communicate(A, B);
     BOOST_CHECK_EQUAL(A.get_configuration().is_joint(), false);
-    BOOST_CHECK_EQUAL(A.get_configuration().current.find(config_member_from_id(B_id))->can_vote, true);
+    BOOST_CHECK_EQUAL(bool(A.get_configuration().current.find(config_member_from_id(B_id))->can_vote), true);
```

Comment:
Couldn't the result be compared directly to the expected value instead of doing the bool conversion?
```suggestion
    BOOST_CHECK_EQUAL(A.get_configuration().current.find(config_member_from_id(B_id))->can_vote, raft::can_vote_t::yes);
```
I think this might lead to more clear messages in case of failure.

Same for the other occurrences in the test.

Reply:
The direct comparisons still not there